src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/OasysService.kt

```diff
@@ -77,10 +92,25 @@ class OasysService(val oasysApiClient: OasysApiClient) {
     return offenceDetail.entity
   }
 
+  fun getRoshFull(assessmentId: Long): OasysRoshFull? {
+    val roshFull = when (val response = oasysApiClient.getRoshFull(assessmentId)) {
+      is ClientResult.Failure -> {
+        log.warn("Failure to retrieve data ${response.toException().cause}")
+        AuthorisableActionResult.Success(null)
+      }
+
+      is ClientResult.Success -> {
+        AuthorisableActionResult.Success(response.body)
+      }
+    }
+
+    return roshFull.entity
+  }
+
   fun getRelationships(assessmentId: Long): OasysRelationships? {
     val relationships = when (val response = oasysApiClient.getRelationships(assessmentId)) {
       is ClientResult.Failure -> {
-        log.warn("Failure to retrieve data")
+        log.warn("Failure to retrieve data ${response.toException().cause}")
```

Comment:
Nice, it's been hard debugging the prisoner search issues without more detailed errors, this'll help that.